### ISSUE TITLE
Feature Request: Add to F-Droid Store

### ISSUE THREAD

**Issue by Perflyst** [BaronKiko/LauncherHijack#24]
Please add this app to the f-droid store: https://f-droid.org/

**Comment by BaronKiko** [comment]
Never heard of f-droid but it looks cool. Will probably do so if I put out another release.
Regardless it's not an issue so I'm closing it.

**Comment by Perflyst** [comment]
You don't need a new release to do this. If it is okay for you I can open the merge request on F-Droid. I just need permission from you to do so because this is how they do it on F-Droid

**Comment by BaronKiko** [comment]
I know a new release isn't needed but it before it goes up I need to understand how f-droid works so I can push updates and the like whereas if I'm to put more time into this project it would first resolve some fire TV issues I was having.

So not to be rude but it's just not up there on priority. I would also like to maintain control of all the permissions so I think it's best I do so myself.

**Comment by BaronKiko** [comment]
Well I spent the past couple hours looking into this and it's not trivial to release. Had to install a linux vm as it's only supported there and the app refuses to upload anything without giving me any feedback.
If you have any experience with this help would be appreciated as I'm simply lost.

**Comment by Perflyst** [comment]
"the app refuses to upload anything without giving me any feedback"
What do you mean with this?

You just need to create a txt file with the metadata of the app and try to build it locally. If succeeded open a merge request. I will try.

**Comment by BaronKiko** [comment]
I have been trying to use there publishing app listed here: https://f-droid.org/en/tutorials/create-repo/

I don't think this will be able to go onto the main repo as I don't believe it's buildable standalone and there is no signing going on. Nor is this repo super well maintained.
What I really want is to just give them an APK and when I have an update I can just give them a new APK.

**Comment by davidfi01** [comment]
Upload it to apkmirror

**Comment by Perflyst** [comment]
Okay so you tried to create a own repo. This is not needed here.
You can submit it to the main repo, it does only update if you create a specific tag. At the moment you have V3, you can set it to check only VX (replace X with number) tags and than release an update on the fdroid store.

You do not need to sign this build, also you dont need to build anything. It will be builded by the fdroid buildserver after the merge.

**Comment by BaronKiko** [comment]
> Upload it to apkmirror

Please don't, if you have to download an APK then you can just get it from here

Signing is more important than you think, it ensures you are getting the code from me and not some random fork. It would take very little work to turn this project into a virus. It's also looking like I will have to make different builds for different platforms which that system may have a hard time with. Having my own repo is the best case but still not what I really want as getting a user to add the repo is a pain. I was planing on having a look at another user suggestion, aptoid, tomorrow as that may be able to do it.

**Comment by Perflyst** [comment]
I understand how important signing is. But what is the issue you have with the main repo from fdroid?

**Comment by BaronKiko** [comment]
I want to be on the main repo, I just want to retain control of building, it's just easier that way. As I can tell with F-Droid if you want to keep that control you simply have to make your own repo. On top of that I don't think I adhere to every section of their inclusion policy: https://f-droid.org/en/docs/Inclusion_Policy/

**Comment by Perflyst** [comment]
You should see it this way: A lot of people trust f-droid an their tools. If you release an apk nobody knows if it is really the source code from this repo here.
Very less people even build the source code. F-Droid does this for the people and they trust it because everything is transparent.

People may not trust you. Why? Because you are e.g. not a known developer.
Why is it easier if you control building? You need to build the app and you need to get trust from users first.

aptoid: Maybe aptoid is easy because you just need to upload the apk and you are done. But how can a user even trust aptoide? It is not open source and not transparent.

**Comment by BaronKiko** [comment]
Your aptoid reasoning is flawed, you can apply the same logic to the official app store and it seems silly. It's just a place for me to point at so users can be notified upon a new release. If they don't trust it then they can get the signed builds directly from here or build it themselves.

If you are worried about security I ask you how many times have you read the source code for something you have downloaded from f-droid? If you are going to that effort then compiling it yourself isn't a concern.

In short regardless of the situation you have to either trust me or read and compile the code yourself.

But I don't really care about who trusts who as this github will always contain the latest releases. My concerns are releasing an unsigned app is just dangerous, and externally controlled build tools are a pain. For me to use f-droid both of those issues would have to be resolved, it's as simple as that.

**Comment by LuccoJ** [comment]
Just for clarity, since reading this issue left me a little confused, although this represents my own understanding and I in no way claim to represent the F-Droid team:

* All the apps in F-Droid's main repository are built by F-Droid buildservers automatically based on provided "recipes".
* The resulting APKs are signed, but they are signed with keys that belong to F-Droid, not with the author's key (which the F-Droid team obviously doesn't hold).
* The one exception is when the build is made reproducible, in which case F-Droid can sign it using the author's own key as it will be identical, but this involves nontrivial steps in the build process.
* I believe this app as it is would not be accepted into F-Droid's main repository, because its license is either not a FOSS license, or its status as such is made unclear by the fact it's a short ad-hoc license written in what aren't really typical legal terms.

**Comment by BaronKiko** [comment]
Ok seems like all the bases are covered. Just sorting these new fire tablet bugs atm. Cheers for all the info.
